src/m_menu.c

```diff
@@ -2775,151 +2728,92 @@ int mult_screens_index; // the index of the current screen in a set
 // to the previous screen. If you leave these off, you can't move from
 // screen to screen.
 
-setup_menu_t keys_settings1[] =  // Key Binding screen strings       
-{
-  {"ACTION"    ,S_SKIP|S_TITLE,m_null,KB_X,M_Y},
-  {"FIRE"        ,S_INPUT     ,m_scrn,KB_X,M_Y+1*M_SPC,{0},input_fire},
-  {"FORWARD"     ,S_INPUT     ,m_scrn,KB_X,M_Y+2*M_SPC,{0},input_forward},
-  {"BACKWARD"    ,S_INPUT     ,m_scrn,KB_X,M_Y+3*M_SPC,{0},input_backward},
-  {"STRAFE LEFT" ,S_INPUT     ,m_scrn,KB_X,M_Y+4*M_SPC,{0},input_strafeleft},
-  {"STRAFE RIGHT",S_INPUT     ,m_scrn,KB_X,M_Y+5*M_SPC,{0},input_straferight},
-  {"USE"         ,S_INPUT     ,m_scrn,KB_X,M_Y+6*M_SPC,{0},input_use},
-  {"RUN"         ,S_INPUT     ,m_scrn,KB_X,M_Y+7*M_SPC,{0},input_speed},
-  {"STRAFE"      ,S_INPUT     ,m_scrn,KB_X,M_Y+8*M_SPC,{0},input_strafe},
-  {"TURN LEFT"   ,S_INPUT     ,m_scrn,KB_X,M_Y+9*M_SPC,{0},input_turnleft},
-  {"TURN RIGHT"  ,S_INPUT     ,m_scrn,KB_X,M_Y+10*M_SPC,{0},input_turnright},
-  {"180 TURN"    ,S_INPUT|S_STRICT,m_scrn,KB_X,M_Y+11*M_SPC,{0},input_reverse},
-
-  {"TOGGLES"     ,S_SKIP|S_TITLE,m_null,KB_X,M_Y+13*M_SPC},
-  {"AUTORUN"     ,S_INPUT     ,m_scrn,KB_X,M_Y+14*M_SPC,{0},input_autorun},
-  {"MOUSELOOK"   ,S_INPUT     ,m_scrn,KB_X,M_Y+15*M_SPC,{0},input_mouselook},
-  {"VERTMOUSE"   ,S_INPUT     ,m_scrn,KB_X,M_Y+16*M_SPC,{0},input_novert},
-
-  // Button for resetting to defaults
-  {0,S_RESET,m_null,X_BUTTON,Y_BUTTON},
+setup_menu_t keys_settings1[] =  // Key Binding screen strings
```

Comment:
These are the best tab names I could think of given the low res 4:3 constraints:

"Key Bindings"
```
1   2   3   4   5   6   7
```

"Weapons"
```
Preferences   Cosmetic
```

"Status Bar / HUD"
```
HUD1   HUD2   Messages
```

The other categories are single page only, so no tabs are needed.

Reply:
So I made quick implementation for General menu:

https://github.com/fabiangreffrath/woof/assets/5077629/ec9384e8-3657-4d98-84aa-8b086d002c6a

Should I continue with this branch?



> "Key Bindings"
> 
> ```
> 1   2   3   4   5   6   7
> ```

Yes, this is an issue. Not sure what to do.

Reply:
😮 Gorgeous!

Reply:
How do you change menu pages with the keyboard?

Reply:
> How do you change menu pages with the keyboard?

At the moment in the same way as before. I don't think we should change it, people are getting used to the PrBoom+ menu.

Reply:
This is an Absolute Masterpiece! ❤️‍🔥 Totally smears that old-fashioned menu on the floor. I can understand what "getting used to" is, but.. Sometimes it is necessary to make a step forward and not hold dear "old traditions that founding fathers formed", maybe? This is change, very big change. But change not "for change" or a tribute to modern years, but change for good in all it's meanings: usability, appearance, style and sense of artistical taste.

Speaking personally, even by watching 7 seconds video was much enough to love this from first look. If there will voting for "use such menu or not", please count my vote as doubled, as I'm raising both hands. 
✋👩🏻🤚

Reply:
> Should I continue with this branch?

I guess we have an answer to this question. 😁

Reply:
> At the moment in the same way as before. I don't think we should change it, people are getting used to the PrBoom+ menu.

I think we can get rid of the PREV/PAGEx/NEXT bottom line then? It should be pretty clear that e.g. pressing right brings us to the next tab to the right.

Reply:
> I think we can get rid of the PREV/PAGEx/NEXT bottom line then? It should be pretty clear that e.g. pressing right brings us to the next tab to the right.

I agree. I will try to implement Ceski's markup exactly: https://github.com/fabiangreffrath/woof/assets/56656010/856ddd6d-4390-47c4-8591-d0f5ad7e609d

Reply:
What is that horizontal line beneath the tab names?

Reply:
> So I made quick implementation for General menu:

That looks great! Is the tab switching instant on mouseover? I'd make it click to switch. I'm curious is some of the other parts of the mockup will work now. That's a very nice initial test.

> > "Key Bindings"
> > ```
> > 1   2   3   4   5   6   7
> > ```
> 
> Yes, this is an issue. Not sure what to do.

You can try the numbers, or something like the numbers with brackets (`[1]` etc.), or if you want names, I think you have to cut a page and do some creative reorganizing that could be controversial (e.g. do we need every one of those cheats?). That's all I could think of.

Reply:
> What is that horizontal line beneath the tab names?

It indicate the selected tab. We'll have to be minimalist without graphics.

Reply:
Maybe an aspect ratio dependent set of names? Maybe that's too much.

Reply:
> It indicate the selected tab. We'll have to be minimalist without graphics.

Yes, I understand that, but what graphics patch is this? The underscore char?

Reply:
> Maybe an aspect ratio dependent set of names? Maybe that's too much.

I think that's too much. I already confused some users with the savegame snapshots being only visible in widescreen mode.

Reply:
> Is the tab switching instant on mouseover? I'd make it click to switch.

I agree, that's a bug 😄 

> I think you have to cut a page and do some creative reorganizing that could be controversial (e.g. do we need every one of those cheats?).

Yeah, maybe we don't need that many inputs. I'll try `[n]` first.

Reply:
> Yes, I understand that, but what graphics patch is this? The underscore char?

Just line. `V_FillRect(x + video.deltaw, rect->y + M_SPC, rect->w, 1, cr_gold[v_lightest_color]);`

Reply:
Another rough idea is if the tab "bar" gets beyond a certain width, trim the right to `> > >`. Then as you gradullly make it to the last tab, the left becomes `< < <`.  The arrows would probably have to be clickable for better usability.

Reply:
> Just line. `V_FillRect(x + video.deltaw, rect->y + M_SPC, rect->w, 1, cr_gold[v_lightest_color]);`

Perfect!

Reply:
Also, the skull seems to naturally feel like a back button now. It may not make sense in practice thoug, so you could try it first.

Reply:
I manage to fit everything: 
![woof0000](https://github.com/fabiangreffrath/woof/assets/5077629/a71ad9ca-f000-4c2f-8ffa-407e7cc6bcd8)

Reply:
Re: sound menu, yes only one please.

Reply:
> Not sure about this one. Overall, I'm not happy with the "Status Bar/HUD" menu. I have to admit that I don't know what most of these options do, and I don't like the Boom HUD. But we can't remove it.

I have some ideas that should make everyone happy but it can be for later. Sneak peek:

```
HUD Type: Crispy, Boom, Boom Full <-- bars
Widescreen HUD: Yes/No
Coords/Stats Layout: Horizontal, Vertical
```

Reply:
> I have some ideas that should make everyone happy but it can be for later. Sneak peek:
> 
> ```
> HUD Type: Crispy, Boom, Boom Full <-- bars
> Widescreen HUD: Yes/No
> Coords/Stats Layout: Horizontal, Vertical
> ```

I like it! Much cleaner.

Reply:
Either way works, but I'll be away from my computer for half a day or so.

Reply:
NO problem, we have plenty of time.

Reply:
Here's an attempt to reorganize the "Status Bar / HUD" menu: https://github.com/ceski-1/woof/commit/ab8ab2dc3aa7e42292bf1aff712624d6282d8754

Screenshots: [HUD](https://github.com/fabiangreffrath/woof/assets/56656010/06a8ec1e-fd91-42ca-b9ff-ad1d4fc4059e), [Widgets](https://github.com/fabiangreffrath/woof/assets/56656010/65b442b4-36e3-4e2a-8fe6-3c8daa366d82)

Reply:
I think there's no easy way to tie everything together nicely in this menu. `hud_active` seems to be a very old setting that now serves a dual purpose. It's the "mode" for a Boom style HUD via WOOFHUD lump, where mode 0 = Minimal, 1 = Compact, 2 = Distrubuted. But then it's also used for Crispy HUD where 0 = Off?, 1 = Normal 4:3, 2 = Widescreen. Maybe the "Mode" strings need to be updated to match the selected HUD type. I don't know.

Reply:
It's beautiful, I like it already. I especially like that the crosshair is in a separate tab.

About `hud_active`, maybe we can refactor it? It would affect people's configs, but it's not a big deal to set it up again (I don't use the fullscreen HUD though).

Reply:
Thanks, would you like to copy it into your branch and tune it further? I'm not sure what to do with `hud_active` to be honest. If it's possible to dynamically change "choice" strings now, maybe have "HUD Mode" update to a new list whenever "HUD Type" changes, like described above.

Reply:
> Thanks, would you like to copy it into your branch and tune it further?

Ok, I will copy it. You can also commit to this branch if you like, although we don't usually do that.

> If it's possible to dynamically change "choice" strings now, maybe have "HUD Mode" update to a new list whenever "HUD Type" changes, like described above.

Yes, it's possible, good idea.

Reply:
> Here's an attempt to reorganize the "Status Bar / HUD" menu:

Yes, I like it.

> I think there's no easy way to tie everything together nicely in this menu. `hud_active` seems to be a very old setting that now serves a dual purpose. It's the "mode" for a Boom style HUD via WOOFHUD lump, where mode 0 = Minimal, 1 = Compact, 2 = Distributed. But then it's also used for Crispy HUD where 0 = Off?, 1 = Normal 4:3, 2 = Widescreen. Maybe the "Mode" strings need to be updated to match the selected HUD type. I don't know.

In the end it figures:

As you pointed out, for the Boom HUD the `hud_active` modes mean 0 = Minimal, 1 = Compact, 2 = Distributed. If you have enabled the Crispy HUD instead, `1` means that the status bar widgets are kept together as they are in the original status bar and `2` means that the status bar widgets are moved to the screen edges in widescreen mode. That's the Crispy HUD's pendant to "distributed mode". This does not affect the Boom HUD widgets which can be additionally shown (stats, time, etc.). These are either aligned at the original 320p screen borders or also moved to the screen edges in widescreen mode depending on the value of the "widescreen arrangement" switch.

Reply:
what about reporpousing the `screenblocks` setting and get rid of the others?
7=normal statusbar
8=crispy hud
9=boom hud (normal)
10=boom hud (distributed)
11=boom hud (minimal)
12=nothing

the the `hud_widescreen_widgets` would be used to switch between the stock/wide layout for both the crispy and the boom huds

the old inter-doom and nugget already do something similar

Reply:
> In the end it figures:

I have created an overview to show what each setting actually means:

https://github.com/fabiangreffrath/woof/wiki/HUD-modes

Reply:
> what about reporpousing the `screenblocks` setting and get rid of the others?

Interesting, but maybe not in this PR?

> I have created an overview to show what each setting actually means:
> 
> https://github.com/fabiangreffrath/woof/wiki/HUD-modes

Thanks! Looks complicated.

Reply:
> the `hud_widescreen_widgets` would be used to switch between the stock/wide layout for both the crispy and the boom huds

The widescreen idea helped me think of a different, but similar improvement, thank you.

@rfomin I'll push something to your branch in a bit.